src/coreComponents/denseLinearAlgebra/denseLASolvers.hpp

```diff
@@ -0,0 +1,349 @@
+ * @tparam SOL_TYPE The type of the solution vector x. Must support indexing with `x[i]`.
+ *
+ * @param[in] A The 3x3 matrix representing the system of equations. Must have size 3x3.
+ * @param[in] b The 3-element vector representing the right-hand side of the equation.
+ * @param[out] x The 3-element vector that will store the solution to the system.
+ */
+template< typename MATRIX_TYPE, 
+          typename RHS_TYPE, 
+          typename SOL_TYPE >
+GEOS_HOST_DEVICE
+inline
+void solveThreeByThreeSystem( MATRIX_TYPE const & A, RHS_TYPE const & b, SOL_TYPE && x)
+{
+    LvArray::tensorOps::internal::checkSizes< 3, 3 >( A );
+    LvArray::tensorOps::internal::checkSizes< 3 >( b );
+    LvArray::tensorOps::internal::checkSizes< 3 >( x );
+
+    real64 const detA = A[0][0] * (A[1][1] * A[2][2] - A[1][2] * A[2][1]) -
+                        A[0][1] * (A[1][0] * A[2][2] - A[1][2] * A[2][0]) +
+                        A[0][2] * (A[1][0] * A[2][1] - A[1][1] * A[2][0]);
+
+    GEOS_ERROR_IF_LT_MSG( LvArray::math::abs(detA), LvArray::NumericLimits< real64 >::epsilon;, "Singular system." );
+
+    real64 const detX0 = b[0] * (A[1][1] * A[2][2] - A[1][2] * A[2][1]) -
+                         b[1] * (A[0][1] * A[2][2] - A[0][2] * A[2][1]) +
+                         b[2] * (A[0][1] * A[1][2] - A[0][2] * A[1][1]);
+
+    real64 const detX1 = A[0][0] * (b[1] * A[2][2] - b[2] * A[2][1]) -
+                         A[0][1] * (b[0] * A[2][2] - b[2] * A[2][0]) +
+                         A[0][2] * (b[0] * A[1][2] - b[1] * A[1][0]);
+
+    real64 const detX2 = A[0][0] * (A[1][1] * b[2] - A[1][2] * b[1]) -
+                         A[0][1] * (A[1][0] * b[2] - A[1][2] * b[0]) +
+                         A[0][2] * (A[1][0] * b[1] - A[1][1] * b[0]);
+
+    x[0] = detX0 / detA;
+    x[1] = detX1 / detA;
+    x[2] = detX2 / detA;
+}
+
+/**
+ * @brief Solves a linear system where the matrix is upper triangular using back substitution.
+ * 
+ * This function solves the linear system `Ax = b`, where `A` is an upper triangular matrix, using 
+ * back substitution. The solution `x` is computed and stored in the provided output vector.
+ * 
+ * @tparam N The size of the square matrix `A`.
+ * @tparam MATRIX_TYPE The type of the matrix `A`.
+ * @tparam RHS_TYPE The type of the right-hand side vector `b`.
+ * @tparam SOL_TYPE The type of the solution vector `x`.
+ * @param[in] A The upper triangular matrix representing the coefficients of the system.
+ * @param[in,out] b The right-hand side vector. It is used to compute the solution, but the values are not preserved.
+ * @param[out] x The solution vector. The result of solving the system `Ax = b` using back substitution.
+ */
+template< std::ptrdiff_t N, 
+          typename MATRIX_TYPE,
+          typename RHS_TYPE,
+          typename SOL_TYPE >
+void solveUpperTriangularSystem( MATRIX_TYPE const & A, RHS_TYPE & b, SOL_TYPE && x )
+{
+  for (std::ptrdiff_t i = N - 1; i >= 0; --i)
```

Comment:
```suggestion
  for (integer i = N - 1; i >= 0; --i)
```

Reply:
I was trying to be consistent with all LvArray methods.